src/components/total-product/Product.jsx

```diff
@@ -3,23 +3,20 @@ import style from './Product.module.css';
 import { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 
-const Product = ({ id, title, img, price, soldOut }) => {
+const Product = ({ id, title, img, price }) => {
   const [heart, setHeart] = useState(false);
   const navigate = useNavigate();
   function onClick() {
     setHeart((cur) => !cur);
   }
-  function onClickDetail() {
-    if (soldOut) {
-      return null;
-    } else {
-      navigate(`/products/${id}`, { state: { id, title, img, price, soldOut } });
-    }
-  }
 
   return (
-    <li className={style.wrap} onClick={onClickDetail}>
-      {soldOut ? <div className={style.soldout}>SOLDOUT</div> : null}
+    <li
+      className={style.wrap}
+      onClick={() => {
+        navigate(`/products/${id}`, { state: { id, title, img, price } });
+      }}
+    >
```

Comment:
```suggestion
    <Link to="/products/${id}" state={{ id, title, img, price }} >
        <li className={style.wrap} >
        ...
```
Link 컴포넌트로 주소를 바꾸고 state도 보낼 수 있습니다.
navigate는 로직내에서 사용하는걸 추천합니다